pygmt/src/meca.py

```diff
@@ -293,9 +293,13 @@ def meca(
         newcols = ["longitude", "latitude", "depth"] + param_conventions[convention]
         if "plot_longitude" in spec.columns and "plot_latitude" in spec.columns:
             newcols += ["plot_longitude", "plot_latitude"]
+            spec[["plot_longitude", "plot_latitude"]] = spec[
+                ["plot_longitude", "plot_latitude"]
+            ].astype(str)
             kwargs["A"] = True
         if "event_name" in spec.columns:
             newcols += ["event_name"]
+            spec["event_name"] = spec["event_name"].astype(str)
         # reorder columns in DataFrame
         spec = spec.reindex(newcols, axis=1)
     elif isinstance(spec, np.ndarray) and spec.ndim == 1:
```

Comment:
Hmm, I was re-reading #1129 again and was gonna say, it's a little surprising that a 2D numpy array passed in with `plot_longitude` and `plot_latitude` columns (stored in float) works (since the `test_meca_spec_2d_array` unit test is passing fine). I guess the GMT C code is smart enough with handling data from `virtualfile_from_matrix` vs `virtualfile_from_vectors`.

Nothing to do here, just a random comment.

Reply:
> it's a little surprising that a 2D numpy array passed in with `plot_longitude` and `plot_latitude` columns (stored in float) works (since the `test_meca_spec_2d_array` unit test is passing fine).

That's a good point. I think it shouldn't work. The `test_meca_spec_2d_array` test passes because both `plot_longitude` and `plot_latitude` are 0 (which means no offsetting due to backward compatibility issues).

I'll find some time and see if offsetting using a 2D numpy array works, and if not, will  try to find a fix.

Reply:
Ah yes, so we should probably update the test to use a non-zero offset. I actually wanted to parse all the user inputs into a `pandas.DataFrame` (which can hold different dtypes like int, str, float) in https://github.com/GenericMappingTools/pygmt/pull/1613#discussion_r744349790 which meant everything goes through `virtualfile_from_vectors`, but that was superseded by #1784 that maintained two separate paths going via `virtualfile_from_vectors` for `dict`/`pd.DataFrame` or `virtualfile_from_matrix` for 2D `np.ndarray` inputs. Just as an idea :slightly_smiling_face:

If this ends up getting too complicated, we can merge in this PR first to fix offsets for dict/pandas inputs, and have a separate one to handle offsets for 2D numpy array inputs.

Reply:
> If this ends up getting too complicated, we can merge in this PR first to fix offsets for dict/pandas inputs, and have a separate one to handle offsets for 2D numpy array inputs.

I prefer to work on it in a separate PR.